docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -1,4 +1,103 @@
+[[java-rest-high-document-search-request]]
+==== Search Request
+
+The `SearchRequest` is used for any operation that has to do with retrieving 
+documents, aggregations, suggestions and also offers ways of requesting 
+highlighting on the resulting documents.
+
+In its most basic form, a query can be added to the request like this: 
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-basic]
+--------------------------------------------------
+
+<1> Creates the `SeachRequest`. Without arguments this runs against all indices.
+<2> Most details about the search can be added to the `SearchSourceBuilder`.
+<3> Add a `match_all` query to the `SearchSourceBuilder`.
+
+When the search is executed, it returns a `SearchResponse` object that serves
+as an entry point to a lot of information about the search response and the
+results:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-response-basic]
+--------------------------------------------------
+
+<1> Actually perform the search. The response contains information about the
+execution of the request itself, like the `took` time or if the request timed
+out.
+<2> Get the `SearchHits` from the response. This object provides access to
+global result parameters like number of total hits or the maximum score.
+<3> Get the individual `SearchHit` objects. These are returned in the order
+determined by the search. `SearchHit` provides access to document details like
+its id or the document `source`.
+
+==== Optional request arguments
+
+Lets first look at some of the optional argument of a `SearchRequest`.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-details]
+--------------------------------------------------
+
+<1> The request can be restricted to one or more indices by passing their names
+to the constructor. 
+<2> The request can also be limited to one or more types.
+<3> Optional: provide a routing parameter for the request.
+<4> Optional: `IndicesOptions` control how unavailable indices are resolved and
+how wildcard expressions are expanded
+<5> Optional: Set the preference to execute the search to prefer local shards.
+Defaults to randomize across shards.
+
+==== Using the SearchSourceBuilder
+
+Most options controlling the search behavior can be set on the `SearchSourceBuilder`, 
+which contains more or less the equivalent of the options in the search request body of the Rest API.
+
+Here are a few examples of some common options:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-source-basics]
```

Comment:
rather than having a big snippet with 5 or more notes, in other docs we did one snippet per option, it is a bit more work but maybe better when visualized and more readable?